src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems: number = 42;
+  const values = [3, 5, 10, 20];
+  const [numItems, setNumItems] = useState(5);
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  let from = 1;
+  let to = numItems;
+
+  if (currentPage && currentPage !== 1) {
+    for (let page = 2; page <= currentPage; page++) {
+      from = to + 1;
+      to = numItems * page;
+
+      if (to > totalItems) {
+        to = totalItems;
+      }
+    }
+  }
```

Comment:
The pagination logic using the for loop is not optimal. It recalculates the 'from' and 'to' values in each iteration which can be simplified. Consider calculating 'from' and 'to' directly based on 'currentPage' and 'numItems', without the need for a loop. For example, 'from' can be calculated as `(currentPage - 1) * numItems + 1` and 'to' as `Math.min(currentPage * numItems, totalItems)`.